vendor/github.com/go-openapi/errors/schema.go

```diff
@@ -65,6 +65,32 @@ const (
 	failedAllPatternPropsNoIn = "%s.%s failed all pattern properties"
 )
 
+// All code responses can be used to differentiate errors for different handling
+// by the consuming program
+const (
+	// CompositeErrorCode remains 422 for backwards-compatibility
+	// and to separate it from validation errors with cause
+	CompositeErrorCode = 422
+	// InvalidTypeCode is used for any subclass of invalid types
+	InvalidTypeCode = 600 + iota
+	RequiredFailCode
+	TooLongFailCode
+	TooShortFailCode
+	PatternFailCode
+	EnumFailCode
+	MultipleOfFailCode
+	MaxFailCode
```

Comment:
In my opinion it is more interesting to know which parameters caused an error, validation constraints, and the expected type. Specific validation criteria that failed validation doesn't make a lot of sense, since I would just display a specific set of conditions for the user that needs to be taken into account rather than pointing what is exactly wrong. I am not saying it is all wrong, I just pointing it is an overkill for me.